### DISCUSSION TITLE
Enforce preference of preincrementation wherever applicable

### DISCUSSION THREAD

**Discussion by Chubercik** [godotengine/godot-proposals#8042] [Engine Core]
Although the performance boost might be negligible (if at all present due to how modern compilers should be able to squeeze out any optimizations possible), wouldn't it make sense to enforce a code quality check for using preincrementation in certain code structures, e.g. `for` loops, where we don't really care about the return value of the `i++` statement?

As it stands, currently there are both `i++` and `++i` statements in Godot's codebase, enforcing one or the other would make it more uniform.

**Comment by Calinou** [comment]
Please check whether this is the case with [Godbolt](https://godbolt.org/) when using a modern GCC or Clang version with release optimizations enabled.

**Reply by Chubercik** [reply]
Fair enough, no difference in the resulting binary; what about code style unification?

**Reply by Calinou** [reply]
`i++` should likely be preferred in this case as it's more commonly used overall, and maps more closely to `i += 1` which is used elsewhere (and in GDScript).
